psi4/driver/p4util/util.py

```diff
@@ -98,10 +99,71 @@ def cubeprop(wfn, **kwargs):
     cp.compute_properties()
 
 
-def set_memory(bytes):
-    """Function to reset the total memory allocation."""
-    core.set_memory(bytes)
+def set_memory(inputval):
+    """Function to reset the total memory allocation. Takes memory value 
+    *inputval* as type int, float, or str; int and float are taken literally
+    as bytes to be set, string taken as a unit-containing value (e.g., 30 mb)
+    which is case-insensitive.
+    
+    :returns: *memory_amount* (float) Number of bytes of memory set
 
+    :examples:
+
+    >>> # [1] Passing absolute number of bytes
+    >>> psi4.set_memory(20000000)
+    >>> psi4.get_memory()
+    Out[1]: 20000000L
+
+    >>> # [2] Passing memory value as string with units
+    >>> psi4.set_memory('30 GB')
+    >>> psi4.get_memory()
+    Out[2]: 30000000000L
+
+    :tests:
+
+    {2000: 2000L,
+     24088624.9: 24088624L,
+     1.0e8: 100000000L,
+     '': AttributeError("'NoneType' object has no attribute 'group'"),
+     '21986': AttributeError("'NoneType' object has no attribute 'group'"),
+     '100 kb': 100000L,
+     '100.0 KiB': 102400L,
+     '500 MB': 500000000L,
+     '2 eb': 2000000000000000000L}
+
+    """
+    # Handle memory given in bytes directly (int or float)
+    if isinstance(inputval, int) or isinstance(inputval, float):
+        val = inputval
+        units = ''
+    # Handle memory given as a string
+    if isinstance(inputval, basestring):
+        memory_string = re.compile(r'^\s*?(\d*\.?\d+)\s+([KMGTPBE]i?B)', re.IGNORECASE)
+        matchobj = re.search(memory_string, inputval)
+        if matchobj:
+            val = float(matchobj.group(1))
+            units = matchobj.group(2)
+        else:
+            raise ValidationError("""Invalid memory specification '{}'.""".format(inputval))
```

Comment:
I'm voting for (and implementing with a similar helpfully TypeError msg) ValidationError since it prints more helpfully